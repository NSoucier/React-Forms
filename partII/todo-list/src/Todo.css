.remove {
    border-radius: 20%;
    border: 1px solid dimgray;
    margin-left: 10px;
    background-color: rgba(255, 0, 0, 0.741);
    color: white;
}