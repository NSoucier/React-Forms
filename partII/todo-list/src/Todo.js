import './Todo.css'

function Todo({ task, remove }) {
    const removeTask = () => {
        remove(task);
    };
    
    return (
        <li>
            {task}
            <button className="remove" onClick={removeTask}>X</button>
        </li>
    )
}

export default Todo;