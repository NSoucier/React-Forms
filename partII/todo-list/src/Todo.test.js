import { render } from "@testing-library/react";
import Todo from "./Todo.js";

// smoke test
it('renders without crashing', function() {
  render(<Todo />);
});

// snapshot test
it('matches snapshot', function() {
    const {asFragment} = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
});