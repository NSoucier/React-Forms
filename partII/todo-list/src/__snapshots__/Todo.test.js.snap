// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li>
    <button
      class="remove"
    >
      X
    </button>
  </li>
</DocumentFragment>
`;
